src/objects.c

```diff
@@ -2930,8 +2930,8 @@ CK_RV p11prov_obj_set_ec_encoded_public_key(P11PROV_OBJ *key,
     }
 
     if (add_attrs > 0) {
-        void *ptr = OPENSSL_realloc(key->attrs, sizeof(CK_ATTRIBUTE) * (key->numattrs + add_attrs));
-        if (!ptr) {
+        key->attrs = OPENSSL_realloc(key->attrs, sizeof(CK_ATTRIBUTE) * (key->numattrs + add_attrs));
+        if (!key->attrs) {
```

Comment:
On failure realloc will leave the original pointer untouched, so the correct way is to use ptr, and only if it is not null then key->attrs = ptr; 

That was the original intention of this code and the bug is the missing overwriting of key->attrs after the null check.

Reply:
Hi @simo5 

Updated the code, thanks for the pointer.